### ISSUE TITLE
Crop black margins from screenshots

### ISSUE THREAD

**Issue by FloWuenne** [napari/napari#6114]
## Motivation

Currently, the screenshot functionality from the command produces screenshots with black margins around the image, making these screenshots not directly useable in any figures etc without cropping.

## Pitch

 It would be nice to add an option to be able to automatically crop the black margins from the image. An example function to do something along those lines could be something like this:
 
 ```
     
def crop_black_margins(img: Image.Image, output_path: str) -> None:
    """
    This function takes a PIL Image as input, detects and crops black margins from the image, 
    and saves the cropped image to a specified path.

    Parameters
    ----------
    image : PIL.Image.Image
        The input image from which to crop black margins.
    output_path : str
        The path at which to save the cropped image.

    Returns
    -------
    None
    """
    image = Image.open(img)

    # Convert image to grayscale for simpler processing
    img_gray = image.convert('L')

    # Convert to numpy array for easier processing
    img_array = np.array(img_gray)

    # Detect left margin
    for margin_left in range(img_array.shape[1]):
        if np.any(img_array[:, margin_left] != 0):
            break

    # Detect right margin
    for margin_right in range(img_array.shape[1]-1, -1, -1):
        if np.any(img_array[:, margin_right] != 0):
            break

    # Detect top margin
    for margin_top in range(img_array.shape[0]):
        if np.any(img_array[margin_top, :] != 0):
            break

    # Detect bottom margin
    for margin_bottom in range(img_array.shape[0]-1, -1, -1):
        if np.any(img_array[margin_bottom, :] != 0):
            break

    # Crop the image
    img_cropped = image.crop((margin_left, margin_top, margin_right, margin_bottom))

    # Save the cropped image
    img_cropped.save(output_path)
    ``` 

## Alternatives

Alternatively, the way screenshots with `canvas_only = TRUE` work could be adjusted to not have black margins to start with.

**Comment by melonora** [comment]
Hi @FloWuenne,

Thanks for opening this issue! Recently, there has been some talk on [forum.image.sc](https://forum.image.sc/t/saving-image-from-napari/50379/14). I think we indeed should look into this. 
As we talked about earlier today and also just for record keeping for the others. One thing that would be a possible addition to a feature like this is to also make sure that a scale_bar can be visualized on top of the data as this is a typical requirement for publication images.
